modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/ContentDashboardManagementToolbarPropsTransformer.js

```diff
@@ -14,252 +14,197 @@
-		openSelectionModal({
-
-			// buttonAddLabel: Liferay.Language.get('select'),
-
-			height: '70vh',
+		},
+		selectFileExtension: {
+			data: {
+				itemValueKey: false,
+				selectEventName: 'selectedFileExtension',
+				selectItemURL: item.data.selectFileExtensionURL,
+				urlParamName: 'fileExtension',
+			},
 			multiple: true,
-			onSelect: (selectedItem) => {
-				if (selectedItem) {
-					const assetTags = selectedItem.map((tag) => tag.value);
-
-					let redirectURL = itemData?.redirectURL;
-
-					assetTags.forEach((assetTag) => {
-						redirectURL = addParams(
-							`${portletNamespace}assetTagId=${assetTag}`,
-							redirectURL
-						);
-					});
-
-					navigate(redirectURL);
-				}
+			size: 'md',
+		},
+		selectScope: {
+			data: {
+				itemValueKey: 'groupid',
+				selectEventName: 'selectedScopeIdItem',
+				selectItemURL: item.data.selectScopeURL,
+				urlParamName: 'scopeId',
 			},
-			selectEventName: `${portletNamespace}selectedAssetTag`,
+			multiple: false,
 			size: 'lg',
-			title: itemData?.dialogTitle,
-			url: itemData?.selectTagURL,
-		});
+		},
 	};
 
-	const selectContentDashboardItemSubtype = (itemData) => {
-		openSelectionModal({
-
-			// buttonAddLabel: Liferay.Language.get('select'),
-
-			height: '70vh',
-			multiple: true,
-			onSelect: (selectedItems) => {
-				let redirectURL = itemData?.redirectURL;
-
-				selectedItems.forEach((item) => {
-					redirectURL = addParams(
-						`${portletNamespace}contentDashboardItemSubtypePayload=${JSON.stringify(
-							item
-						)}`,
-						redirectURL
-					);
-				});
-
-				navigate(redirectURL);
-			},
-			selectEventName: `${portletNamespace}selectedContentDashboardItemSubtype`,
-			size: 'md',
-			title: itemData?.dialogTitle,
-			url: itemData?.selectContentDashboardItemSubtypeURL,
-		});
+	return {
+		...item,
+		data: {
+			...item.data,
+			...options[item.data.action].data,
+		},
+		multiple: options[item.data.action].multiple,
+		size: options[item.data.action].size,
 	};
+};
 
-	const selectFileExtension = (itemData) => {
-		openSelectionModal({
+// /FAKE
 
-			// buttonAddLabel: Liferay.Language.get('select'),
+const DEFAULT_VALUES = {
+	buttonAddLabel: Liferay.Language.get('select'),
+	iframeBodyCssClass: '',
+	modalHeight: '70vh',
+};
 
-			height: '70vh',
-			multiple: true,
-			onSelect: (selectedItems) => {
-				let redirectURL = itemData?.redirectURL;
+/**
+ * Returns true if the specified value is an object. Not arrays, native events or functions.
+ * @param {?} value Variable to test.
+ * @return {boolean} Whether variable is an object.
+ */
+const _isObjectStrict = (value) =>
+	typeof value === 'object' &&
+	!Array.isArray(value) &&
+	value !== null &&
+	!Object.hasOwn(value, 'currentTarget');
 
-				selectedItems.forEach((item) => {
-					redirectURL = addParams(
-						`${portletNamespace}fileExtension=${item}`,
-						redirectURL
-					);
-				});
+/**
+ * Returns URL with proper search params.
+ */
+const _buildParams = ({
```

Comment:
Ei @beltranrengifo , great work refactoring all this code. 👏  👏  👏  👏 

In this function _buildParams I think that it is not very clear what is the difference between item and selection arguments. Possible solutions that I see:

- Use another name for those args
- Receive all that info in just one arg or refactor the args
- A comment explaining the args

Reply:
totally agree @claraizquierdo 
this part is messy, I'm redoing the url build, let you know in a while
thxs!!!

Reply:
@claraizquierdo 
I think is clearer now, [this commit](https://github.com/liferay-tango/liferay-portal/pull/2679/commits/1796a104c8a5a91493ad92f399fa99c75886418b)

What do you think?